module3/homework/advanced_calculator/advancedCalculator.cpp

```diff
@@ -0,0 +1,142 @@
+#include "advancedCalculator.hpp"
+
+#include <algorithm>
+#include <cmath>
+#include <functional>
+#include <map>
+#include <sstream>
+
+std::map<char, std::function<double(double, double)>> operation = {
+    {'+', std::plus<double>()},
+    {'-', std::minus<double>()},
+    {'*', std::multiplies<double>()},
+    {'/', std::divides<double>()},
+    {'^', [](auto first, auto second) { return pow(first, second); }},
+    {'$', [](auto first, auto second) { return pow(first, 1.0 / second); }},
+    {'%', std::modulus<int>()},
+    {'!', [](auto first, auto second) {
```

Comment:
It's good that you put the factorial there, but I think the code would be more readable if you implemented this function elsewhere